CONTRIBUTING.md

```diff
@@ -20,14 +20,84 @@
 - Avoid adding third-party dependencies, extra files, extra headers, etc.
 - Always consider cross-compatibility with other operating systems and architectures
 - Avoid fancy-looking modern STL constructs, use basic `for` loops, avoid templates, keep it simple
-- There are no strict rules for the code style, but try to follow the patterns in the code (indentation, spaces, etc.). Vertical alignment makes things more readable and easier to batch edit
+- Vertical alignment makes things more readable and easier to batch edit
 - Clean-up any trailing whitespaces, use 4 spaces for indentation, brackets on the same line, `void * ptr`, `int & a`
-- Naming usually optimizes for common prefix (see https://github.com/ggerganov/ggml/pull/302#discussion_r1243240963)
+- Use sized integer types in the public API
+- Declare structs with `struct foo {}` instead of `typedef struct foo {} foo`
+    - In C++ code omit the `struct` keyword whenever it is not necessary
+    > [!NOTE]
+    > This guideline is yet to be applied to the `llama.cpp` codebase. New code should follow this guideline.
+- Try to follow the existing patterns in the code (indentation, spaces, etc.). In case of doubt use `clang-format` to format the added code
```

Comment:
Should we explicitly instruct contributors to use 4 spaces for indentation?

Reply:
It's already in the guidelines:

> Clean-up any trailing whitespaces, use 4 spaces for indentation, brackets on the same line, void * ptr, int & a